enhancements/microshift/dummy-gateway-interface-ovn.md

```diff
@@ -0,0 +1,458 @@
+---
+title: dummy-gateway-interface-ovn
+authors:
+- "@pliurh"
+reviewers:
+- "@jcaamano, OpenShift Networking"
+- "@pacevedom, Microshift"
+- "@pmtk, MicroShift"
+- "@trozet, OpenShift Networking"
+- "@zshi-redhat, MicroShift"
+approvers:
+- "@dhellmann"
+api-approvers:
+- None
+creation-date: 2023-04-26
+last-updated: 2023-04-27
+tracking-link:
+- https://issues.redhat.com/browse/NP-654
+see-also:
+- None
+replaces:
+- None
+superseded-by:
+- None
+---
+
+# Single Node OVN-Kubernetes for Microshift
+
+## Summary
+
+This enhancement proposes a new external gateway configuration of OVN-Kubernetes
+for Microshift. With the new configuration, a dummy interface will be created
+and the `br-ex` bridge will use it as the gateway interface.
+
+## Motivation
+
+Microshift is designed to run on IoT and edge devices, it can be deployed on
+various kinds of devices with different network environment. 
+
+Today configure-ovs.sh (microshift-ovs-init.service) sets up OVN-Kubernetes
+gateway bridge `br-ex` for OVN-Kubernetes.  There has been several types of
+issues associated with it mainly due to:
+
+1. transient network interruption when it moves the node IP from uplink to
+   gateway bridge
+
+2. certain type of network devices are not supported (e.g. wifi, tap/tun)
+
+3. it relies on (default) route info
+
+### User Stories
+
+* As an Microshift cluster administrator, I want the installation experience of
+Microshift is similar as a normal application. I don't want to have any
+networking interrupt during the installation.
+
+* As an Microshift cluster administrator, I want to use a specific network
+  devices (such as wifi, tap/tun) as the node default interface.
+
+* As an Microshift cluster administrator, I want to install Microshift on an
+  isolated host that has no default gateway available.
+
+### Goals
+
+* Decouple the br-ex bridge from the host physical interfaces.
+
+### Non-Goals
+
+* Creating a special gateway mode in OVN-Kubernetes for single node cluster.
```

Comment:
Note that there's an effort to create relocatable SNO.
Similar to the goal of this enahancment relocatable SNO requires decoupling between br-ex bridge and the host physical interface because it will move between networks.
Though the way we currently allow SNO relocation is somewhat simpler (but still relay on ovn-configuration.service) by adding the internal IP to the external `br-ex` bridge used by OVS. so when the node get relocated to another network while the external network changes the internal IP is still configured on the bridge. 
I think we can align to the solution proposed in this enhancement.

Reply:
Yeah, if we have a similar use case, it would be great to have an aligned design. 

@trozet suggested using the `br-ex` bridge without binding any host physical interface. The traffic flow should be almost the same as what I described in this document. And I'll update the enhancement accordingly. But it can only work with local GW mode. I'm making a patch at https://github.com/ovn-org/ovn-kubernetes/pull/3577.